### PR TITLE
20230816-release-1v5-the-documentation-release

### PR DESCRIPTION
**Pull request by douzzer**
See `ChangeLog.md` and git log for change details.

No new major features in this commit or this release, but lots of minor bug fixes, and a mountain of documentation.

### CODE REVIEW

**Review by JacobBarthelmeh — CHANGES_REQUESTED** [review]
'make check' on older Ubuntu 18

```
$ make check
In file included from /home/jak/Documents/wolfsentry/wolfsentry/wolfsentry.h:221:0,
                 from /home/jak/Documents/wolfsentry/src/wolfsentry_internal.h:26,
                 from /home/jak/Documents/wolfsentry/src/wolfsentry_util.c:24:
/home/jak/Documents/wolfsentry/wolfsentry/wolfsentry_settings.h:727:5: error: initializer element is not constant
     __wolfsentry_config
     ^~~~~~~~~~~~~~~~~~~
/home/jak/Documents/wolfsentry/wolfsentry/wolfsentry_settings.h:727:5: note: (near initialization for ‘wolfsentry_build_settings.config’)
Makefile:253: recipe for target 'wolfsentry-dynamic-build-test/src/wolfsentry_util.So' failed
make[1]: *** [wolfsentry-dynamic-build-test/src/wolfsentry_util.So] Error 1
/home/jak/Documents/wolfsentry/Makefile.analyzers:417: recipe for target 'dynamic-build-test' failed
```

`make check` on MacOSX

```
$ make check
test_all_shared(80978,0x10b99edc0) malloc: *** error for object 0x7f9d0c609690: pointer being freed was not allocated
test_all_shared(80978,0x10b99edc0) malloc: *** set a breakpoint in malloc_error_break to debug
/bin/bash: line 1: 80978 Abort trap: 6           LD_LIBRARY_PATH="/var/folders/3h/76pvsw3s1y31dpmfb741svqm0000gn/T//wolfsentry-dynamic-build-test" "/var/folders/3h/76pvsw3s1y31dpmfb741svqm0000gn/T//wolfsentry-dynamic-build-test/tests/$test" > /dev/null
make: *** [dynamic-build-test] Error 134

```

**Comment by JacobBarthelmeh** [comment]
The MacOSX make check could potentially be ignored. The old Mac env. testing on is potentially having issues with linking static libraries. The old Ubuntu make check though should work unless there is a minimum compiler version requirement for wolfSentry?

File: src/routes.c [hunk 1]
```diff
@@ -419,20 +423,39 @@ static wolfsentry_errcode_t wolfsentry_route_init(
...
 
-    memset(new, 0, offsetof(struct wolfsentry_route, data));
+#ifdef WOLFSENTRY_ROUTE_INIT_TEST_FOR_UNINITED_DATA
+    /* don't initialize bytes in the allocation past the end of the address data. */
+    if (new_size > offsetof(struct wolfsentry_route, data) + (size_t)data_addr_offset + WOLFSENTRY_BITS_TO_BYTES(remote->addr_len) + WOLFSENTRY_BITS_TO_BYTES(local->addr_len))
+        new_size = offsetof(struct wolfsentry_route, data) + (size_t)data_addr_offset + WOLFSENTRY_BITS_TO_BYTES(remote->addr_len) + WOLFSENTRY_BITS_TO_BYTES(local->addr_len);
+    for (int testbyte = 1, uninited_offset = 0;; ++testbyte) {
+        byte *p;
+        memset(new, testbyte, new_size);
+    #if 0
```
[the remark this answers is not in the document]
Comment (douzzer) [reply src/routes.c:433]: yup there is.

File: src/routes.c [hunk 2]
```diff
@@ -460,8 +483,27 @@ static wolfsentry_errcode_t wolfsentry_route_init(
         }
     }
 
-    new->header.refcount = 1;
-    new->header.id = WOLFSENTRY_ENT_ID_NONE;
+#ifdef WOLFSENTRY_ROUTE_INIT_TEST_FOR_UNINITED_DATA
+    #if 0
```
[the remark this answers is not in the document]
Comment (douzzer) [reply src/routes.c:487]: nope the gated-out close and open curlies are just to keep the editor from trying to indent the code in between the `_UNINITED_DATA` sections.

**Comment by douzzer** [comment]
> The MacOSX make check could potentially be ignored. The old Mac env. testing on is potentially having issues with linking static libraries. The old Ubuntu make check though should work unless there is a minimum compiler version requirement for wolfSentry?

Whoops, yeah, for Mac it's `make check-macosx`, which we do want to pass but it's separate from the linux-centric `check`.

The Ubuntu18 issue we do need to resolve -- I may need to resort to properly gated gnuisms to make it DTRT (mainly `__builtin_constant_p`).

**Review by JacobBarthelmeh — APPROVED** [review]

**Pull request merged**